### DISCUSSION TITLE
Scrolling for particular raw only

### DISCUSSION THREAD

**Discussion by Sheetal-Rathod** [VasilKalchev/LiquidMenu#80] [Q&A] [answered]
Hello,
I have lcd of (20,4) and I just want to scroll only for raw 1,2,3 want to skip line 0.,how can i do it?

#include <LiquidCrystal.h>
#include <LiquidMenu.h>
#include <Key.h>
#include <Keypad.h>

LiquidCrystal lcd(44,43, 42, 40, 39, 38, 37); 
char *CT_String[] = {"[0]","[10]","[50]","[75]","[100]","[300]","[500]","[&UP]"};
const byte ROWS = 4; //four rows
const byte COLS = 5; //three columns
char keys[ROWS][COLS] = 
{
  {'A','B','C','D','E'},
  {'F','G','H','I','J'},
  {'K','L','M','N','O'},
  {'P','Q','R','S','T'} 
};
byte rowPins[ROWS] = {40,39,38,37}; //connect to the row pinouts of the keypad A4,A5,A6,A7
byte colPins[COLS] = {19,7,51,52,27}; //connect to the column pinouts of the keypad D1,E7,F6,F7,G1

Keypad keypad = Keypad( makeKeymap(keys), rowPins, colPins, ROWS, COLS );

LiquidLine CTScreen_line(0, 0, "ANG   CT   DIA  ");
LiquidLine CTScreen_line1(6,1, CT_String[0]);
LiquidLine CTScreen_line2(6,1, CT_String[1]);
LiquidLine CTScreen_line3(6,1, CT_String[3]);
LiquidLine CTScreen_line4(6,1, CT_String[4]);
LiquidLine CTScreen_line5(6,1, CT_String[5]);
LiquidLine CTScreen_line6(6,1, CT_String[6]);
//LiquidLine CTScreen_line6(6, 1, CT_String[7]);

LiquidScreen CTScreen;
LiquidMenu menu(lcd);
 void blankFunction() 
 {
    return;
 }

void keypadEvent(KeypadEvent key)
{
    DDRA = DDRA | B11110000; 
    switch (keypad.getState()) 
    {
      case PRESSED:
              switch(key)
              {
                  case 'R': // 'ENT'
                               break;

                 case 'P': // 'UP'
                           //lcd.clear();
                            menu.switch_focus(true);
                                
                                break; 
                 case 'Q': // 'DOWN'
                           menu.switch_focus(false);
                         
                                 break;                            
              }  
                break;                 
      case RELEASED:break; 
      case HOLD:  break;
      
    }
}
void setup() 
{
  lcd.begin(20, 4);
  keypad.setDebounceTime(200);
  keypad.addEventListener(keypadEvent);
  
  CTScreen.add_line(CTScreen_line);
  CTScreen.add_line(CTScreen_line1);
  CTScreen.add_line(CTScreen_line2);
  CTScreen.add_line(CTScreen_line3); 
  CTScreen.add_line(CTScreen_line4);
  CTScreen.add_line(CTScreen_line5);
  CTScreen.add_line(CTScreen_line6);

  CTScreen.set_displayLineCount(3);
  menu.add_screen(CTScreen);
  
  CTScreen_line1.set_focusPosition(Position::LEFT);
  CTScreen_line2.set_focusPosition(Position::LEFT);
  CTScreen_line3.set_focusPosition(Position::LEFT);
  CTScreen_line4.set_focusPosition(Position::LEFT);
  CTScreen_line5.set_focusPosition(Position::LEFT);
  CTScreen_line6.set_focusPosition(Position::LEFT);

  CTScreen_line1.attach_function(1, blankFunction);
  CTScreen_line2.attach_function(1, blankFunction);
  CTScreen_line3.attach_function(1, blankFunction);
  CTScreen_line4.attach_function(1, blankFunction);
  CTScreen_line5.attach_function(1, blankFunction);
  CTScreen_line6.attach_function(1, blankFunction);

  menu.update();
}
void loop()
{
    DDRA = DDRA | B00000000; 
    char key = keypad.getKey();  
    delay(100);
}

can you please review this code? give me some idea?

**Comment by veit1337** [comment, marked as answer]
I assume that `LiquidLine CTScreen_line(0, 0, "ANG CT DIA ");` is your "headline". I also haven't figured out a way to scroll only one line, but with a work-around it is possible to "simulate" that only one line is scrolling. Just add the headline right before every scroll-line. 

```
CTScreen.add_line(CTScreen_line);
CTScreen.add_line(CTScreen_line1);
CTScreen.add_line(CTScreen_line);
CTScreen.add_line(CTScreen_line2);
CTScreen.add_line(CTScreen_line);
CTScreen.add_line(CTScreen_line3);
CTScreen.add_line(CTScreen_line);
CTScreen.add_line(CTScreen_line4);
CTScreen.add_line(CTScreen_line);
CTScreen.add_line(CTScreen_line5);
CTScreen.add_line(CTScreen_line);
CTScreen.add_line(CTScreen_line6);
```


The headline doesn't have a attached function. This means that when you switch focus (scroll), 2 lines get scrolled at once and you end up with the headline above the scroll-line again. 
This is the work-around I use to only "scroll" one line on the display.
I hope this is helpful.
